### PR TITLE
Check MicroCeph for API errors

### PR DESCRIPTION
**Pull request by masnax**
Potential solution for #427 

Seems this isn't an issue with 22.04 necessarily, but with how we are using the microceph client. For the [`AddDisk`](https://github.com/canonical/microceph/blob/e32ad499ad83dcf6401891cc5739524bcdcd6dd4/microceph/client/disks.go#L18-L29) function, API side errors are compiled in `types.DiskAddResponse` instead of the error returned by the client function. 

As such, we were ignoring most disk-add errors. The reason this led to hangs is because we would proceed to create the LXD storage pools, which would hang forever on Ceph trying to create OSD pools without any OSDs. 

In particular, if encryption is requested, MicroCeph requires the following to be run before running `microcloud init`:
```bash
sudo snap connect microceph:dm-crypt
sudo snap restart microceph.daemon
```

We link to this in the MicroCeph docs in our initialization steps [here](canonical-microcloud.readthedocs-hosted.com/en/latest/microcloud/how-to/initialise/#interactive-configuration) where it mentions encryption.

In my testing, if the above is run before following the steps in the issue, then MicroCloud completes setup successfully. Otherwise, it hangs.

### CODE REVIEW

File: cmd/microcloud/main_init.go [hunk 1]
```diff
@@ -763,10 +763,16 @@ func (c *initConfig) setupCluster(s *service.Handler) error {
...
 				logger.Debug("Adding disk to MicroCeph", logger.Ctx{"name": name, "disk": disk.Path})
-				_, err = cephClient.AddDisk(context.Background(), client, &disk)
+				resp, err := cephClient.AddDisk(context.Background(), client, &disk)
 				if err != nil {
 					return err
 				}
+
+				for _, report := range resp.Reports {
+					if report.Error != "" {
+						return fmt.Errorf("Failed to add disk to MicroCeph: %s", report.Error)
```
Comment (roosterfish) [inline cmd/microcloud/main_init.go:773]: Do we have to concatenate the errors from multiple `Reports`?

**Review by roosterfish — APPROVED** [review]
Good catch!

**Comment by roosterfish** [comment]
> In particular, if encryption is requested, MicroCeph requires the following to be run before running

Can it be that this is auto connected sometimes? Because I remember doing some MicroCloud deployments in the past with Ceph encryption and I have never manually did the connection with `dm-crypt`.

**Comment by tomponline** [comment]
> As such, we were ignoring most disk-add errors.

Ah good that was my hunch, so glad we dug into it to find this. Thanks

**Comment by roosterfish** [comment]
> Can it be that this is auto connected sometimes?

Nvm the test suite is doing this that's why it works https://github.com/canonical/microcloud/blob/main/test/includes/microcloud.sh#L1215.

**Review by gabrielmougard — APPROVED** [review]

File: cmd/microcloud/main_init.go (hunk 1 above)
Comment (masnax) [reply cmd/microcloud/main_init.go]: Fixed.

**Review by roosterfish — APPROVED** [review]

**Pull request merged**